tests/rancher/rancher_k3s_worker.pm

```diff
@@ -0,0 +1,54 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Web browser UI test for rancher container
+# Maintainer: George Gkioulis <ggkioulis@suse.com>
+
+use base 'x11test';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use utils;
+use mm_network;
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    mutex_wait('support_server_ready');
+    configure_hostname(get_required_var('HOSTNAME'));
+    configure_dhcp();
+
+    script_retry("nslookup gitlab.com", delay => 3, retry => 10);
+
+    assert_script_run("curl -L https://get.k3s.io -o ~/get_k3s", 360);
+    assert_script_run("chmod +rx ~/get_k3s");
+
+    assert_script_run "ssh-keygen -t rsa -P '' -C '`whoami`@`hostname`' -f ~/.ssh/id_rsa";
+    assert_script_run('ssh-keyscan master >> ~/.ssh/known_hosts');
+    exec_and_insert_password('ssh-copy-id root@master');
+
+    barrier_wait('k3s_prepared');
+
+    script_retry('ssh root@master cat /var/lib/rancher/k3s/server/node-token', delay => 10, retry => 30, timeout => 90);
```

Comment:
maybe we should use something like [wait_for_ssh](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/instance.pm#L206) instead ?

Reply:
I'm actually not waiting for SSH but for the cluster to generate the `node-token` which is then used by the other nodes.
```suggestion
    script_retry('ssh root@master test -f /var/lib/rancher/k3s/server/node-token', delay => 10, retry => 30, timeout => 90);
```

Reply:
Removed this line as the `master` already checks for this file before approaching the `cluster_prepared` barrier.